kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java

```diff
@@ -793,9 +796,16 @@ protected void handleProduceRequest(KafkaHeaderAndRequest produceHar,
                     return;
                 }
 
-                final Consumer<PersistentTopic> persistentTopicConsumer = persistentTopic ->
-                        publishMessages(persistentTopic, byteBuf, numMessages, validRecords, fullPartitionName,
-                                offsetConsumer, errorsConsumer);
+                final Consumer<PersistentTopic> persistentTopicConsumer = persistentTopic -> {
+                    // check system topic
+                    if (persistentTopic.isSystemTopic()) {
+                        log.error("Not support produce message to system topic. topic: {}", persistentTopic);
+                        throw new InvalidTopicException(Errors.INVALID_TOPIC_EXCEPTION.message());
+                    }
```

Comment:
It's better to put this logic o `publishMessages`. First, `publishMessages` is responsible to check if `persistentTopic` is null. 

And you shouldn't thrown exceptions here, I've already check if it's the system topic. See https://github.com/streamnative/kop/blob/0e3ef0de20f5e28c2a7adc7b6396f30f72c55920/kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java#L698-L702

Reply:
Ok, i remove the check code.